src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -253,9 +261,10 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 
             NamedTypeSymbol? namedType = null;
 
-            ImmutableArray<NamedTypeSymbol> namespaceOrTypeMembers;
+            Debug.Assert(scope is NamespaceSymbol or NamedTypeSymbol);
             bool isTopLevel = scope.IsNamespace;
```

Comment:
this is one of the methods that benefits hugely from not allocating temp strings.